src/main/java/org/sagebionetworks/web/client/widget/entity/EntityMetadata.java

```diff
@@ -94,4 +109,30 @@ public void clear() {
 		view.clear();
 	}
 	
+	 public void configureStorageLocation(String containerEntityId) {
+		 synapseClient.getUploadDestinations(containerEntityId, new AsyncCallback<List<UploadDestination>>() {
```

Comment:
Could be more explicit about the view state here.  That is, instead of relying on view.clear to hide the upload destination panel, this method could initially set visibility to false, and then show if it successfully determines the destination.